patterns/README.md

```diff
@@ -1030,3 +1030,363 @@ A field MAY have a `missingValues` property that MUST be an `array` where each e
+   value label and attach it to the field. As with (1), provide option to skip
+   automatically dropping values specified in the `missingValues` property and
+   instead add them in order to the end of the value label, encoded using
+   extended missing values if supported.
+
+3. Although none of Stata, SAS or SPSS currently permits designating a specific
+   variable as ordered, Stata permits attaching arbitrary metadata to
+   individual variables. Thus, in cases where the `enumOrdered` property is
+   present, this information can be stored in Stata to inform the analyst and
+   to prevent loss of information when generating Frictionless data packages
+   from within Stata.
+
+#### Software that supports categoricals or factors (Pandas, R, Julia)
+
+1. In cases where a field has an `enum` constraint but no `enumLabels`
+   property, automatically define a categorical or factor using the `enum`
+   values in order, and convert the variable to categorical or factor type
+   using this definition. Provide option to skip automatically dropping values
+   specified in the `missingValues` property and instead add them in order to
+   the end of the `enum` values when defining the categorical or factor.
+
+2. In cases where the data are stored in encoded form (e.g., as integers) and
+   a corresponding `enumLabels` property is present, translate the data using
+   the `enumLabels` object, define a categorical or factor using the values of
+   the `enumLabels` object in lexical order of the keys, and convert the
+   variable to categorical or factor type using this definition. Provide
+   option to skip automatically dropping values specified in the
+   `missingValues` property and instead add them to the end of the
+   `enumLabels` values when defining the categorical or factor.
+
+3. In cases where a field has an `enumOrdered` property, use that when
+   defining the categorical or factor.
+
+#### All software
+
+Although the extensions proposed here are intended primarily to support the
+use of value labels and categoricals in software that supports them, they also
+provide additional functionality when reading data into any software that can
+handle tabular data. Specifically, the `enumLabels` property may be used to
+support the use of enums even in cases where value labels or categoricals are
+not being used. For example, it is standard practice in software for analyzing
+genetic data to code sex as 0, 1 and 2 (corresponding to "Unknown", "Male" and
+"Female") and affection status as 0, 1 and 2 (corresponding to "Unknown",
+"Unaffected" and "Affected"). In such cases, the `enumLabels` property may be
+used to confirm that the data follow the standard convention or to indicate
+that they deviate from it; it may also be used to translate those codes into
+human-readable values, if desired.
+
+### Notes
+
+This pattern originally included a proposal to add an optional field-specific
```

Comment:
```suggestion
While this pattern is designed as an extension to [Table Schema](https://specs.frictionlessdata.io/table-schema/) fields, it could also be used to document `enum` values of properties in [profiles](https://specs.frictionlessdata.io/profiles/), such as contributor roles.

This pattern originally included a proposal to add an optional field-specific
```